### ISSUE TITLE
Crash when executing a symlink

### ISSUE THREAD

**Issue by Dennisbonke** [managarm/mlibc#93]
When executing a symlink (in my test case, `/usr/bin/cc` pointing to `/usr/bin/gcc`) the following is printed
```
In function operator(), file ../../../src/mlibc/options/posix/generic/posix_stdlib.cpp:200
__ensure(!S_ISLNK(st.st_mode) && "TO
```
The full ensure line is as follows
```
__ensure(!S_ISLNK(st.st_mode) && "TODO: Use readlink in realpath()");
```

**Comment by avdgrinten** [comment]
This should be reasonably easy to fix: "just" call readlink when a symlink is encountered, append it to the current path (or replace the current path if it is absolute) and continue resolving the path.

**Comment by Dennisbonke** [comment]
Fixed by 7f0938e09e3230f303ebc0f37df84bbef5f46407
